domain/src/test/java/org/oppia/android/domain/platformparameter/PlatformParameterSingletonTest.kt

```diff
@@ -0,0 +1,164 @@
+package org.oppia.android.domain.platformparameter
+
+import android.app.Application
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.google.common.truth.Truth.assertThat
+import dagger.BindsInstance
+import dagger.Component
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.oppia.android.app.model.PlatformParameter
+import org.oppia.android.util.platformparameter.PlatformParameterValue
+import org.robolectric.annotation.Config
+import org.robolectric.annotation.LooperMode
+import javax.inject.Inject
+import javax.inject.Singleton
+
+private const val STRING_PLATFORM_PARAMETER_NAME = "string_platform_parameter_name"
+private const val STRING_PLATFORM_PARAMETER_VALUE = "string_platform_parameter_value"
+
+private const val INTEGER_PLATFORM_PARAMETER_NAME = "integer_platform_parameter_name"
+private const val INTEGER_PLATFORM_PARAMETER_VALUE = 1
+
+private const val BOOLEAN_PLATFORM_PARAMETER_NAME = "boolean_platform_parameter_name"
+private const val BOOLEAN_PLATFORM_PARAMETER_VALUE = true
+
+private const val INCORRECT_PLATFORM_PARAMETER_NAME = "incorrect_platform_parameter_name"
+
+/**
+ * [PlatformParameterSingletonTest] verifies the working of [PlatformParameterSingleton] by testing
+ * the [PlatformParameterValue] received in different cases
+ * */
+@RunWith(AndroidJUnit4::class)
+@LooperMode(LooperMode.Mode.PAUSED)
+@Config(application = PlatformParameterSingletonTest.TestApplication::class)
+class PlatformParameterSingletonTest {
+
+  @Inject
+  lateinit var platformParameterSingleton: PlatformParameterSingleton
+
+  private val mockPlatformParameterMap by lazy {
+    val stringPlatformParameter = PlatformParameter.newBuilder()
+      .setString(STRING_PLATFORM_PARAMETER_VALUE).build()
+    val integerPlatformParameter = PlatformParameter.newBuilder()
+      .setInteger(INTEGER_PLATFORM_PARAMETER_VALUE).build()
+    val booleanPlatformParameter = PlatformParameter.newBuilder()
+      .setBoolean(BOOLEAN_PLATFORM_PARAMETER_VALUE).build()
+
+    mapOf(
```

Comment:
mention the type to prevent `Type inference` warnings. 

Official reference - https://kotlinlang.org/spec/type-inference.html

Reply:
I have made the change, but just confirming that Android Studio was not giving any warnings on this point. So was this a manual check thing or something which can be detected automatically in some way

Reply:
oh, It was giving a warning for me, `red line` under the `mapOf()`